### DISCUSSION TITLE
Customizing your devTools: replacing 'anonymous' with function names

### DISCUSSION THREAD

**Discussion by AlexeevSasha** [pmndrs/zustand#2061] [Ideas]
I created a library that defaults to displaying the name of the called function instead of 'anonymous' in devtools.
[zustand-devtools](https://www.npmjs.com/package/zustand-devtools) here, you can find more examplesю.

Do you think there is a chance for my pull request with this feature to be approved? Is such functionality needed?

Thank you for the review, and I wish everyone a great mood!

**Comment by dai-shi** [comment]
It sounds like a great idea.

Hmm, but `getCallerFunctionNameFromStack` looks like a trick that we may want to avoid in the zustand library code.

Is it a well known technique? You can open a PR to discuss further if you don't mind if it won't be merged in the end.

**Reply by AlexeevSasha** [reply]
Yeah, it's not exactly pretty, but unfortunately I haven't found a better method to get the name of the function

**Comment by dai-shi** [comment]
btw, we are interested in this technique in Jotai too.
- https://github.com/pmndrs/jotai/issues/931
- https://github.com/jotaijs/jotai-devtools/discussions/55
